src/workload.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Workload manager."""
+
+from typing import Dict, Optional, Tuple
+from ops import pebble
+
+
+# A manifest is a 2-tuple of (files, layer), where "files" is a mapping from path to contents.
+# If contents is None, it means deletion.
+# TODO: contents could also be bytes
+Manifest = Tuple[Optional[Dict[str, Optional[str]]], Optional[pebble.Layer]]
+
+
+class ManifestUpdateError(Exception):
+    """Raised for layer update errors."""
+
+
+class Workload:
+    """Partial amalgamation of ops.Container and pebble."""
```

Comment:
I don't understand why we can't just use `Container` here. As far as I can tell, it has all the methods we need: `can_connect`, `remove_path`, `push`, `get_plan`, `add_layer`, and `replan`. Am I missing something?

So I'd rather this were just a stand-alone function that took a container and a manifest, something like:

```python
def apply_manifest(manifest: Manifest, container: ops.Container) -> None:
    ...
```

Or maybe even an `apply` method on the `Manifest` data class that took just the container instance.